L1Trigger/L1THGCal/test/calib/testCalibration.cc

```diff
@@ -0,0 +1,159 @@
+// -*- C++ -*-
+//
+// Package:    Analyzers
+// Class:      testCalibration
+// 
+/**\class testCalibration testCalibration.cc HGCPFLab/Analyzers/plugins/testCalibration.cc
+
+   Description: [one line class summary]
+
+   Implementation:
+   [Notes on implementation]
+*/
+//
+// Original Author:  Luca Mastrolorenzo
+//         Created:  Mon, 07 Dec 2015 16:38:57 GMT
+// $Id$
+//
+//
+
+
+// system include files
+#include <memory>
+
+// user include files
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/Framework/interface/EDAnalyzer.h"
+#include "FWCore/Utilities/interface/InputTag.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+#include "FWCore/Framework/interface/ESHandle.h"
+#include "FWCore/ServiceRegistry/interface/Service.h"
+#include "DataFormats/Common/interface/Handle.h"
+#include "DataFormats/L1THGCal/interface/HGCalTriggerCell.h"
+#include "CommonTools/UtilAlgos/interface/TFileService.h"
+#include "L1Trigger/L1THGCal/interface/HGCalTriggerBackendProcessor.h"
+#include "TTree.h"
+
+//
+// class declaration
+//
+using namespace std;
+using namespace edm;
+using namespace reco;
+using namespace HGCalTriggerBackend;
+using namespace l1t;
+
+class testCalibration : public edm::EDAnalyzer {
+
+public:
+ 
+    explicit testCalibration(const edm::ParameterSet& );
+    
+    ~testCalibration();
+  
+private:
+ 
+    edm::Service<TFileService> fs_;  
+    virtual void analyze(const edm::Event&, const edm::EventSetup&);
+  
+    // ----------member data ---------------------------
+    EDGetTokenT<HGCalTriggerCellBxCollection > tokenTrgCell_;
+   
+    bool debug_;
+    int _nEvent=0;
+
+    TH1D *h_photon_pt_;
+    TH1D *h_tc_hwPt_;
+    TH1D *h_tc_E_;
+    TH1D *h_tc_pt_;
+    TH1D *h_tc_eta_;
+    TH1D *h_tc_phi_;
+    TH1D *h_tc_layer_;
+
+    TTree *mytree_;  
+
+    double _TC_pt=0.;
+};
+
+
+
+// constructors and destructor
+testCalibration::testCalibration(const edm::ParameterSet& iConfig) : 
+    tokenTrgCell_( consumes< HGCalTriggerCellBxCollection >( iConfig.getParameter<InputTag> ( "triggerCellInputTag" ) ) ),
+    debug_( iConfig.getParameter<bool>( "DebugCode" ) )
```

Comment:
I already commented on that. Use the edm::LogDebug for debug output, and not this debug flag.